lib/checkassert.cpp

```diff
@@ -120,13 +120,21 @@ void CheckAssert::assignmentInAssertError(const Token *tok, const std::string& v
 // checks if side effects happen on the variable prior to tmp
 void CheckAssert::checkVariableAssignment(const Token* assignTok, const Scope *assertionScope)
 {
+    if (!assignTok->isAssignmentOp() && assignTok->tokType() != Token::eIncDecOp)
+        return;
+
     const Variable* prevVar = assignTok->previous()->variable();
```

Comment:
This check uses old code that should use the AST better. I suggest:

    const Variable *var = assignTok->astOperand1()->variable();

That should work for both assignments and pre/post increments

Reply:
however if you want to handle more complex assigned expressions you need some additional code. For example: `assert(*p=0)`, `assert(buf[10]=0)` etc. I still suggest using AST for such cases. You can leave out that in this PR if you want and focus on the "easy" cases.